### PR TITLE
#26: Add interface frame to control camera

### LINKED ISSUE #26: Add interface frame to control camera
Choosing camera type etc.

### PR DESCRIPTION
**Pull request by TransientTetra**

### CODE REVIEW

File: src/view/camera_control_frame.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	{
+		dynamic_cast<CameraView*>(view)->setCameraPosition(glm::vec3(xpos, ypos, zpos));
+	}
+
+	ImGui::Columns(1);
+	ImGui::NewLine();
+	float tempF = mouseSensitivity;
+	ImGui::SliderFloat("Mouse sensitivity", &mouseSensitivity, .1f, 1.f);
+	if (tempF != mouseSensitivity)
+		dynamic_cast<CameraView*>(view)->setCameraSensitivity(mouseSensitivity);
```
Comment (szymonzygzula) [inline src/view/camera_control_frame.cpp:78]: I couldn't spot any difference between the lowest and the highest mouse sensivity. I'd recommend widenning scope of manipulation

File: include/view/camera_view.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	CameraView(Application *application, Window *window);
+
+	void draw() override;
+
+	void setRenderingMode(RenderingMode renderingMode);
+
+	void setCamera(CameraType cameraType);
+	void setCameraPosition(glm::vec3 position);
+	const glm::vec3& getCameraPosition() const;
+	void setCameraSpeed(float cameraSpeed);
```
Comment (szymonzygzula) [inline include/view/camera_view.hpp:27]: Why camera manipulation frame does not include setting camera speed? It could be useful

**Review by szymonzygzula — CHANGES_REQUESTED** [review]
- possibility of hiding this frame is cool. Maybe it would be a good idea to implement this option also in the other frame
- besides comments everything works great

File: include/view/camera_view.hpp (hunk 2 above)
Comment (TransientTetra) [reply include/view/camera_view.hpp:27]: Sure I forgot to add the slider, added it now

File: src/view/camera_control_frame.cpp (hunk 1 above)
Comment (TransientTetra) [reply src/view/camera_control_frame.cpp:78]: Both mouse sensitivity and movement speed are not connected from the frame to camera as the camera internals will be reworked. Both sliders will have to be connected when the rework is done.

**Review by szymonzygzula — APPROVED** [review]
Ok

**Pull request merged**